packages/frontend/src/pages/album/[album].tsx

```diff
@@ -7,50 +7,46 @@ import React, { useEffect, useState } from "react";
 import type { Sequencer, Tag, Song } from "@prisma/client";
 import AlbumSongList from "@/components/dynamic/album/AlbumSongList";
 import { useUser } from "@auth0/nextjs-auth0/client";
+import Custom404 from "../404";
 
 type SongType = Song & { tags: Tag[]; file_sequencer: Sequencer[] };
 
 const Album = () => {
   const { isLoading, user } = useUser();
   const router = useRouter();
-  const [songList, setSongList] = useState<SongType[]>([]);
   const [filteredSongList, setFilteredSongList] = useState<SongType[]>();
   const [albumCoverImages, setAlbumCoverImages] = useState<Song[]>();
   const [gridCol, setGridCol] = useState("");
   const [loading, setLoading] = useState(true);
+  const [notFound, setNotFound] = useState(false);
 
   useEffect(() => {
-    void (async () => {
-      await fetch("/api/song", {
-        method: "GET",
-      })
-        .then(async (res) => {
-          await res
-            .json()
-            .then((result: SongType[]) => {
-              setSongList(result);
-              setLoading(false);
-            })
-            .catch((err) => console.error(err));
+    if (router.query.album) {
+      console.log(router.query.album);
+      void (async () => {
+        await fetch(`/api/album?album=${router.query.album?.toString()}`, {
+          method: "GET",
         })
-        .catch((err) => console.error(err));
-    })();
-  }, []);
-
-  useEffect(() => {
-    if (songList.length > 0 && router.query.album) {
-      const albumExist = songList.some((song) => {
-        return (
-          song.album?.toLowerCase().trim().replace(/ /g, "-") ===
-          router.query.album
-        );
-      });
-
-      if (!albumExist) {
-        void router.push("/404");
-      }
+          .then(async (res) => {
+            await res
+              .json()
+              .then((result: SongType[]) => {
+                if (result && result.length > 0) {
+                  setFilteredSongList(result);
+                } else {
+                  setNotFound(true);
+                }
+                setLoading(false);
+              })
+              .catch((err) => {
+                setNotFound(true);
```

Comment:
change in future, `NotFound` should not be a result in a catch block as it could also be other reasons the fetch failed.